solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -1788,6 +1788,16 @@ public void unregister(String coreName, CoreDescriptor cd, boolean removeCoreFro
     }
     CloudDescriptor cloudDescriptor = cd.getCloudDescriptor();
     if (removeCoreFromZk) {
+      //extra handling for PRS, we need to write the PRS entries from this node directly,
```

Comment:
where is removes entry from state.json file ?

Reply:
Line 1801 will send a message to overseer for the delete core which should remove from state.json. Letting overseer handle state.json but individual nodes handle PRS.